src/core/exporter.js

```diff
@@ -0,0 +1,104 @@
+// module: exporter
+// used in ui/save-html
+
+import { removeReSpec } from "core/utils";
+import { pub } from "core/pubsubhub";
+
+export function serialize(format = "html", doc = document) {
+  const cloneDoc = doc.cloneNode(true);
+  cleanup(cloneDoc);
+  let result = "";
+  switch (format) {
+    case "xml":
+      result = new XMLSerializer().serializeToString(cloneDoc);
+      break;
+    default: {
+      if (cloneDoc.doctype) {
+        result += new XMLSerializer().serializeToString(cloneDoc.doctype);
+      }
+      result += cloneDoc.documentElement.outerHTML;
+    }
+  }
+  return result;
+}
+
+export async function exportDocument(format, mimeType, doc = document) {
+  await doc.respecIsReady;
+  const dataURL = toDataURL(serialize(format, doc), mimeType);
+  const encodedString = dataURL.replace(/^data:\w+\/\w+;charset=utf-8,/, "");
+  const decodedString = decodeURIComponent(encodedString);
+  return decodedString;
+}
+
+export function toDataURL(data, mimeType = "text/html") {
+  const encodedString = encodeURIComponent(data);
+  return `data:${mimeType};charset=utf-8,${encodedString}`;
+}
+
+// Create and download an EPUB 3 version of the content
+// Using (by default) the EPUB 3 conversion service set up at labs.w3.org/epub-generator
+// For more details on that service, see https://github.com/iherman/respec2epub
+export function makeEPubHref() {
```

Comment:
I don't think this is very useful as an export, tbh. This is very closely tied to the Save UI and I don't expect us to use it in other places. Thus, this should probably go back  into ui/save-html